test/integration/service_switcher_ft_tests.js

```diff
@@ -0,0 +1,171 @@
+'use strict'
+
+const request = require('supertest')
+const nock = require('nock')
+
+require('../test_helpers/serialize_mock.js')
+const userCreator = require('../test_helpers/user_creator.js')
+const getApp = require('../../server.js').getApp
+const paths = require('../../app/paths.js')
+const session = require('../test_helpers/mock_session.js')
+const gatewayAccountId = '15486734'
+
+let app
+
+const connectorMock = nock(process.env.CONNECTOR_URL)
+const CONNECTOR_ACCOUNT_PATH = '/v1/frontend/accounts/' + gatewayAccountId
+const LEDGER_TRANSACTION_PATH = '/v1/transaction/{transactionId}?account_id=' + gatewayAccountId
+const LEDGER_EVENTS_PATH = '/v1/transaction/{transactionId}/event?gateway_account_id=' + gatewayAccountId
+const ADMINUSERS_USERS_PATH = '/v1/api/users'
+const ledgerMock = nock(process.env.LEDGER_URL)
+const adminUsersMock = nock(process.env.ADMINUSERS_URL)
+
+function whenGetServiceSwitchTransactionDetails (transactionId, serviceId, baseApp) {
+  return request(baseApp)
+    .get(paths.generateRoute(paths.transactions.serviceSwitchDetail, { serviceId: serviceId, chargeId: transactionId }))
+    .set('Accept', 'application/json')
+}
+
+function ledgerTransactionPathFor (transactionId) {
+  return LEDGER_TRANSACTION_PATH.replace('{transactionId}', transactionId)
+}
+
+function ledgerEventPathFor (transactionId) {
+  return LEDGER_EVENTS_PATH.replace('{transactionId}', transactionId)
+}
+
+describe('The transaction view service switcher scenarios', function () {
+  afterEach(function () {
+    nock.cleanAll()
+    app = null
+  })
+
+  beforeEach(function (done) {
+    let permissions = 'transactions-details:read'
+    let user = session.getUser({
+      gateway_account_ids: [gatewayAccountId],
+      permissions: [{ name: permissions }]
+    })
+    app = session.getAppWithLoggedInUser(getApp(), user)
+
+    userCreator.mockUserResponse(user.toJson(), done)
+    connectorMock.get(CONNECTOR_ACCOUNT_PATH)
+      .reply(200, {
+        'payment_provider': 'sandbox',
```

Comment:
can you use a fixture from `gateway_account_fixtures` to construct this response.